jerry-core/ecma/operations/ecma-function-object.c

```diff
@@ -1068,33 +1056,53 @@ ecma_op_bound_function_get_argument_list (ecma_object_t *func_obj_p, /**< bound
  *         Returned value must be freed with ecma_free_value
  */
 static ecma_value_t JERRY_ATTR_NOINLINE
-ecma_op_function_call_bound (ecma_object_t *func_obj_p, /**< Function object */
-                             const ecma_value_t *arguments_list_p, /**< arguments list */
-                             ecma_length_t arguments_list_len) /**< length of arguments list */
+ecma_op_function_call_bound (ecma_call_args_t *call_args_p) /**< call arguments */
 {
-  JERRY_ASSERT (ecma_get_object_type (func_obj_p) == ECMA_OBJECT_TYPE_BOUND_FUNCTION);
+  JERRY_ASSERT (ecma_get_object_type (call_args_p->func_obj_p) == ECMA_OBJECT_TYPE_BOUND_FUNCTION);
 
   JERRY_CONTEXT (status_flags) &= (uint32_t) ~ECMA_STATUS_DIRECT_EVAL;
 
   ecma_collection_t *bound_arg_list_p = ecma_new_collection ();
   ecma_collection_push_back (bound_arg_list_p, ECMA_VALUE_EMPTY);
 
-  ecma_object_t *target_obj_p = ecma_op_bound_function_get_argument_list (func_obj_p, bound_arg_list_p);
-
-  ecma_collection_append (bound_arg_list_p, arguments_list_p, arguments_list_len);
+  ecma_call_args_t call_args;
+  call_args.func_obj_p = ecma_op_bound_function_get_argument_list (call_args_p->func_obj_p, bound_arg_list_p);
 
+  ecma_collection_append (bound_arg_list_p, call_args_p->argv, call_args_p->argc);
   JERRY_ASSERT (!ecma_is_value_empty (bound_arg_list_p->buffer_p[0]));
 
-  ecma_value_t ret_value = ecma_op_function_call (target_obj_p,
-                                                  bound_arg_list_p->buffer_p[0],
-                                                  bound_arg_list_p->buffer_p + 1,
-                                                  (ecma_length_t) (bound_arg_list_p->item_count - 1));
+  call_args.this_value = bound_arg_list_p->buffer_p[0];
+  call_args.argv = bound_arg_list_p->buffer_p + 1;
+  call_args.argc = (ecma_length_t) (bound_arg_list_p->item_count - 1);
+
+  ecma_value_t ret_value = ecma_op_function_call (&call_args);
 
   ecma_collection_destroy (bound_arg_list_p);
 
   return ret_value;
 } /* ecma_op_function_call_bound */
 
+/**
+ * Create a new function call packed info from the given function call arguments
+ *
+ * @return ecma_call_args_t structure
+ */
+ecma_call_args_t
+ecma_op_function_make_args (ecma_object_t *func_obj_p, /**< Function object */
+                            ecma_value_t this_arg_value, /**< 'this' argument's value */
+                            const ecma_value_t *arguments_list_p, /**< arguments list */
+                            ecma_length_t arguments_list_len) /**< length of arguments list */
+{
+  ecma_call_args_t call_args =
```

Comment:
This structure doesn't look nice alignment-wise. I think `argc` and `argv` should be swapped.